persistent-mongoDB/test/EmbedTestMongo.hs

```diff
@@ -1,5 +1,6 @@
 {-# OPTIONS_GHC -fno-warn-unused-binds -fno-warn-orphans -O0 #-}
 {-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
This change requires `UndecidableInstances`. I give a nice error message for upgrading.